edocumentation-service/src/main/java/org/dcsa/edocumentation/service/BookingService.java

```diff
@@ -81,15 +72,18 @@ private void createDeepObjectsForBooking(BookingTO bookingRequest, Booking booki
     shipmentLocationService.createShipmentLocations(bookingRequest.shipmentLocations(), booking);
   }
 
-  private void createShipmentEvent(Booking booking) {
-    shipmentEventRepository.save(ShipmentEvent.builder()
-      .documentTypeCode(DocumentTypeCode.CBR)
-      .eventClassifierCode(EventClassifierCode.ACT)
-      .documentID(booking.getId())
-      .shipmentEventTypeCode(bookingMapper.toShipmentEventTypeCode(booking.getDocumentStatus()))
-      .documentReference(booking.getCarrierBookingRequestReference())
-      .eventDateTime(booking.getBookingRequestCreatedDateTime())
-      .eventCreatedDateTime(OffsetDateTime.now())
-      .build());
+  @Transactional
+  public Optional<BookingRefStatusTO> cancelBooking(String carrierBookingRequestReference,
+                                                    String reason) {
+    Booking booking = bookingRepository.findBookingByCarrierBookingRequestReference(
+        carrierBookingRequestReference
+      ).orElse(null);
+    if (booking == null) {
```

Comment:
Why not use the Optional you get from the look up? Seems weird to convert it into null and then back to Optional.
The logic here would fit neatly into Optional.map

Reply:
It is not so neat to use `Optional.map`, because I need to keep track of the event and the booking at the same time.